src/Microsoft.AspNet.Mvc.Core/Filters/ProducesAttribute.cs

```diff
@@ -31,7 +32,13 @@ public override void OnResultExecuting([NotNull] ResultExecutingContext context)
 
             if (objectResult != null)
             {
-                SetContentTypes(objectResult.ContentTypes);
+                // Check if FormatFilter has already set the content type
```

Comment:
The comment is not accurate, it does not _has_ to be a format filter. 
suggest rewording it
```If the user or a filter has already set the content type, do not override it.```
nit: lets not break lines if we do not need to.